stdlib/public/runtime/Exclusivity.cpp

```diff
@@ -222,6 +228,20 @@ class AccessSet {
     swift_unreachable("access not found in set");
   }
 
+  /// Return the parent access of \p childAccess in the list.
+  Access *findParentAccess(Access *childAccess) {
```

Comment:
I need to take another look at this. I am a little worried that I may not handle the corner case where childAccess is the first case. I just need to think about it after I eat.